mlxtend/feature_selection/column_selector.py

```diff
@@ -75,12 +76,22 @@ def transform(self, X, y=None):
 
         """
 
-        # We use the loc accessor if the input is a pandas dataframe
-        if hasattr(X, 'loc'):
-            t = X.loc[:, self.cols].values
+        # We use the loc or iloc accessor if the input is a pandas dataframe
+        if hasattr(X, 'loc') or hasattr(X, 'iloc'):
+            if type(self.cols) == tuple:
+                self.cols = list(self.cols)
+            types = {type(i) for i in self.cols}
+            if len(types) > 1:
+                raise ValueError('Elements in `cols` should be all of the same data type.')
+            elif isinstance(self.cols[0], int):
+                t = X.iloc[:, self.cols].values
+            elif isinstance(self.cols[0], str):
+                t = X.loc[:, self.cols].values
```

Comment:
I don't think this should be an elif here (can you please change it back) because we want to check for both things

* first that the column names have the same type
* if the above is true, the second check will check whether these are integer column types.